### ISSUE TITLE
NET-1736 New Rule S1068: Unused "private" fields should be removed

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#7406]
The rule's functionality is already implemented as part of S1144, but we should raise [S1068 ](https://sonarsource.github.io/rspec/#/rspec/S1068) for fields, to be more precise.

```csharp
class MyClass {
  private int foo = 42;  // Noncompliant: foo is unused

  public:
    int compute(int a) {
      return a * 42;
    }
};
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1736](https://sonarsource.atlassian.net/browse/NET-1736)

[NET-1736]: https://sonarsource.atlassian.net/browse/NET-1736?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
